modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb

```
@@ -38,16 +38,23 @@ def update_status
       private
 
       def update_data(form_uuid, file_names, status, case_id)
-        ivc_forms = forms_query(form_uuid, file_names)
+        # First get the query that defines which records we want to update
+        ivc_forms = if file_names.any? { |name| name.end_with?('_merged.pdf') }
+                      # Use all forms for this UUID if it's a merged PDF case
+                      fetch_forms_by_uuid(form_uuid)
+                    else
+                      # Only use specified files for non-merged cases
+                      forms_query(form_uuid, file_names)
+                    end
 
         if ivc_forms.any?
-          ivc_forms.each { |form| form.update!(pega_status: status, case_id:) }
+          ivc_forms.update_all(pega_status: status, case_id:) # rubocop:disable Rails/SkipsModelValidations
```

Is there a particular need for this change? Without the validation checks we could run into a silent failure if the updated data isn't good for whatever reason

---

Good call - no, I don't think so.  I'll revert back to `each`.